packages/core/src/new-api/types/deployer.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * The result of a deployment run.
+ *
+ * @beta
+ */
+export type DeploymentResult =
+  | {
+      status: "success";
+      contracts: Record<
+        string,
+        { contractName: string; contractAddress: string }
+      >;
+    }
+  | {
+      status: "failed" | "hold";
```

Comment:
This needs more info.

Reply:
Tracked in https://github.com/NomicFoundation/ignition/issues/258